### PR TITLE
Add support for generic type constraints in codegen

### PR DESCRIPTION
**Pull request by Carlm-MS**
for scenarios where a generic grain type also has type constraints, the codegen file has compilation errors because it doesn't preserve the constraints. The case where we hit the issue:

```csharp
public interface IPageWorkerGrain<TCollection, TData> : IGrainWithStringKey where TCollection : ICollection<TData>

public class PageWorkerGrain<TCollection, TData> : Grain, IPageWorkerGrain<TCollection, TData> where TCollection: ICollection<TData>, new()
```

### CODE REVIEW

File: src/OrleansCodeGenerator/CodeGeneratorCommon.cs [hunk 1]
```diff
@@ -310,5 +310,39 @@ public static string GetGeneratedNamespace(Type type, bool randomize = false)
 
             return result;
         }
+
+        public static TypeParameterConstraintClauseSyntax[] GetGenericTypeParameterConstraints(Type type)
```
Comment (ReubenBond) [inline src/OrleansCodeGenerator/CodeGeneratorCommon.cs]: This seems to serve the same purpose as `SyntaxFactoryExtensions.GetTypeConstraintSyntax`

**Comment by ReubenBond** [comment]
Happy to see people diving into the codegen code!

LGTM, once the comment about duplicate functionality is addressed.
If you could add your class as a test case, that would be fantastic :smile: 

Thank you

**Comment by gabikliot** [comment]
Worth adding a test.

**Comment by Carlm-MS** [comment]
test added.
removed redundant method.

**Comment by ReubenBond** [comment]
lgtm, thank you!

**Comment by sergeybykov** [comment]
:+1:

**Pull request merged**